node/core/dispute-coordinator/src/dummy.rs

```diff
@@ -0,0 +1,252 @@
+	pub fn concluded_against(self, now: Timestamp) -> DisputeStatus {
+		match self {
+			DisputeStatus::Active => DisputeStatus::ConcludedAgainst(now),
+			DisputeStatus::ConcludedFor(at) =>
+				DisputeStatus::ConcludedAgainst(std::cmp::min(at, now)),
+			DisputeStatus::ConcludedAgainst(at) =>
+				DisputeStatus::ConcludedAgainst(std::cmp::min(at, now)),
+		}
+	}
+
+	/// Whether the disputed candidate is possibly invalid.
+	pub fn is_possibly_invalid(&self) -> bool {
+		match self {
+			DisputeStatus::Active | DisputeStatus::ConcludedAgainst(_) => true,
+			DisputeStatus::ConcludedFor(_) => false,
+		}
+	}
+
+	/// Yields the timestamp this dispute concluded at, if any.
+	pub fn concluded_at(&self) -> Option<Timestamp> {
+		match self {
+			DisputeStatus::Active => None,
+			DisputeStatus::ConcludedFor(at) | DisputeStatus::ConcludedAgainst(at) => Some(*at),
+		}
+	}
+}
+
+async fn run<Context>(subsystem: DisputeCoordinatorSubsystem, mut ctx: Context)
+where
+	Context: overseer::SubsystemContext<Message = DisputeCoordinatorMessage>,
+	Context: SubsystemContext<Message = DisputeCoordinatorMessage>,
+{
+	loop {
+		let res = run_until_error(&mut ctx, &subsystem).await;
+		match res {
+			Err(e) => {
+				e.trace();
+
+				if let Error::Subsystem(SubsystemError::Context(_)) = e {
+					break
+				}
+			},
+			Ok(()) => {
+				tracing::info!(target: LOG_TARGET, "received `Conclude` signal, exiting");
+				break
+			},
+		}
+	}
+}
+
+async fn run_until_error<Context>(
+	ctx: &mut Context,
+	_: &DisputeCoordinatorSubsystem,
+) -> Result<(), Error>
+where
+	Context: overseer::SubsystemContext<Message = DisputeCoordinatorMessage>,
+	Context: SubsystemContext<Message = DisputeCoordinatorMessage>,
+{
+	let mut state = State {};
+
+	loop {
+		match ctx.recv().await? {
+			FromOverseer::Signal(OverseerSignal::Conclude) => return Ok(()),
+			FromOverseer::Signal(OverseerSignal::ActiveLeaves(_)) => {},
+			FromOverseer::Signal(OverseerSignal::BlockFinalized(_, _)) => {},
+			FromOverseer::Communication { msg } => handle_incoming(ctx, &mut state, msg).await?,
+		}
+	}
+}
+
+async fn handle_incoming(
+	_: &mut impl SubsystemContext,
+	_: &mut State,
+	message: DisputeCoordinatorMessage,
+) -> Result<(), Error> {
+	match message {
+		DisputeCoordinatorMessage::ImportStatements { .. } => { /* just drop confirmation */ },
+		DisputeCoordinatorMessage::RecentDisputes(tx) => {
+			let _ = tx.send(Vec::new());
+		},
+		DisputeCoordinatorMessage::ActiveDisputes(tx) => {
+			let _ = tx.send(Vec::new());
+		},
+		DisputeCoordinatorMessage::QueryCandidateVotes(_, tx) => {
+			let _ = tx.send(Vec::new());
+		},
+		DisputeCoordinatorMessage::IssueLocalStatement(_, _, _, _) => {},
+		DisputeCoordinatorMessage::DetermineUndisputedChain {
+			base_number,
+			block_descriptions,
+			tx,
+		} => {
+			let undisputed_chain = block_descriptions
+				.last()
+				.map(|e| (base_number + block_descriptions.len() as BlockNumber, e.block_hash));
+
+			let _ = tx.send(undisputed_chain);
```

Comment:
Please double check, that this is correct.

Reply:
Arithmetic looks correct - chain of 1 with base B would return B+1 with hash(B+1)

Reply:
Is correct iff the first element in the `block_descriptions` has block number `base_number`, which holds by impl of the current `relay_chain_selection`.